### PR TITLE
Ibis: add geospatial support for bigquery

### PR DESCRIPTION
**Pull request by stephencpope**
Add supported geospatial operations for the bigquery backend.

### CODE REVIEW

File: ibis/bigquery/compiler.py [hunk 1]
```diff
@@ -388,6 +396,82 @@ def _formatter(translator, expr):
...
+        ops.GeoNPoints: unary("ST_NumPoints"),
+        ops.GeoPerimeter: multi_arity("ST_Perimeter", 1, 2),
+        ops.GeoPoint: fixed_arity("ST_GeogPoint", 2),
+        ops.GeoTouches: fixed_arity("ST_Touches", 2),
+        ops.GeoUnaryUnion: unary("ST_Union_Agg"),
+        ops.GeoUnion: multi_arity("ST_Union", 1, 2),
+        ops.GeoWithin: fixed_arity("ST_Within", 2),
+        ops.GeoX: unary("ST_X"),
+        ops.GeoY: unary("ST_Y"),
+        # Missing Geospatial ops:
```
Comment (jmileson) [inline ibis/bigquery/compiler.py:437]: Most of these aren't implemented in BQ, this note should indicate that these cannot be supported by BQ:

ST_AsEWKB
ST_AsEWKT
ST_Azimuth
ST_AsGML
ST_AsKML
ST_AsRaster
ST_AsSVG
ST_AsTWKB
ST_Buffer
ST_Crosses
ST_DFullyWithin
ST_Dump
ST_DumpPoints
ST_Envelope
ST_GeometryN
ST_GeometryType
ST_GeomFromEWKB
ST_GeomFromEWKT
ST_GeomFromText
ST_IsValid
ST_LineLocatePoint
ST_LineMerge
ST_LineSubstring
ST_OrderingEquals
ST_Overlaps
ST_Simplify
ST_SRID
ST_SetSRID
ST_Transform

These can be supported and maybe should be added right now:

ST_CoveredBy

ST_DISTANCE is by default on a sphere, so we should just duplicate the definition of `GeoDistance` for ST_Distance_Sphere

File: ibis/bigquery/client.py [hunk 2]
```diff
@@ -152,6 +164,27 @@ def _find_scalar_parameter(expr):
...
 
+def _maybe_to_geodataframe(df, schema):
+    """
+    If the required libraries for geospatial support are installed, and if a
+    geospatial column is present in the dataframe, convert it to a
+    GeoDataFrame.
+    """
+
+    def to_shapely(row, name):
+        return shapely.wkt.loads(row[name]) if row[name] is not None else None
```
Comment (jmileson) [inline ibis/bigquery/client.py:175]: It's so lame that BQ doesn't just do this with the rows already.  I'd kind of prefer that it just unmarshall the shapely objects in the cursor, but maybe we can come back to that later.

File: ibis/bigquery/compiler.py (hunk 1 above)
Comment (stephencpope) [reply ibis/bigquery/compiler.py:438]: Yes I just missed those last to. As to the way the rest are commented; I simply copied exactly the format from the sqlalchemy/PG support.

File: ibis/bigquery/client.py (hunk 2 above)
Comment (stephencpope) [reply ibis/bigquery/client.py:175]: Again, I am copying the prior art from the PG implementation here. Also, as we know, there's an issue in that the geography is coming back spherical...

File: ibis/bigquery/compiler.py (hunk 1 above)
Comment (stephencpope) [reply ibis/bigquery/compiler.py:438]: Actually, there is no ops type connected to ST_Distance_Sphere. The comment list of unimplemented stuff in some cases are things that PostGIS doesn't have and in other cases are things that PostGIS has but ibis doesn't have. ST_Distance_Sphere is one of the latter.

**Pull request merged**